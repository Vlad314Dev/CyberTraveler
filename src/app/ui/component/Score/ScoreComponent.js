import './ScoreStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

class ScoreComponent extends PureComponent
{
    static propTypes = {
        score: PropTypes.number.isRequired
    };

    render()
    {
        const {
            score
        } = this.props;

        return (
            <div block="Score" elem="Wrapper">
                <span block="Score" elem="Value">{ `Score ${ score }` }</span>
            </div>
        );
    }
}

export default ScoreComponent;
