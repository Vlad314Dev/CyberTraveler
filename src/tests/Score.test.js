/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import ScoreComponent from '../app/ui/component/Score/ScoreComponent';

describe('MainMenuTitle', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <ScoreComponent 
                score={ 0 }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <ScoreComponent 
                score={ 0 }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
