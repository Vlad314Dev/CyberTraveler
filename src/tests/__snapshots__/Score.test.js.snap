// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuTitle Should render layout 1`] = `
Array [
  <div
    className="Score__Wrapper"
  >
    <span
      className="Score__Value"
    >
      Score 0
    </span>
  </div>,
]
`;
